### ISSUE TITLE
Fails to build with GCC 5.0: logical not is only applied to the left hand side of comparison

### ISSUE THREAD

**Issue by ppisar** [dsully/perl-crypt-openssl-x509#39]
The -Werror compiler option makes all warnings fatal. GCC 5.0 emits a new warning:

```
gcc -c  -I/usr/include/openssl -I/usr/local/include/ssl -I/usr/local/ssl/include -D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -Wall -Werror   -DVERSION=\"1.806\" -DXS_VERSION=\"1.806\" -fPIC "-I/usr/lib64/perl5/CORE"   X509.c
X509.xs: In function ‘hv_exts’:
X509.xs:184:12: error: logical not is only applied to the left hand side of comparison [-Werror=logical-not-parentheses]
   if ( ! c > 0 ) {
            ^
X509.xs: In function ‘XS_Crypt__OpenSSL__X509_extension’:
X509.xs:871:10: error: logical not is only applied to the left hand side of comparison [-Werror=logical-not-parentheses]
   if (!c > 0) {
          ^
cc1: all warnings being treated as errors
Makefile:347: recipe for target 'X509.o' failed
```

These look like genuine errors. At least from semantic point of view. For example:

```
  c = X509_get_ext_count(x509);
  [...]
  if ( ! c > 0 ) {
    croak("No extensions found\n");
  }
```

**Comment by ppisar** [comment]
Proposed fix:

```
From e1117f13ef3895a738d2e6edfc288175b3677f77 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 11 Feb 2015 13:18:38 +0100
Subject: [PATCH] Fix condition negation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

GCC 5.0 correctly warned on missing parathneses in (! c > 0)
expression. This patch fixes it.

<https://github.com/dsully/perl-crypt-openssl-x509/issues/39>
<https://bugzilla.redhat.com/show_bug.cgi?id=1190816>

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 X509.xs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/X509.xs b/X509.xs
index 23f7666..32a1425 100644
--- a/X509.xs
+++ b/X509.xs
@@ -181,7 +181,7 @@ static HV* hv_exts(X509* x509, int no_name) {
   sv_2mortal((SV*)RETVAL);
   c = X509_get_ext_count(x509);
 
-  if ( ! c > 0 ) {
+  if (!(c > 0)) {
     croak("No extensions found\n");
   }
 
@@ -868,7 +868,7 @@ extension(x509, i)
 
   c = X509_get_ext_count(x509);
 
-  if (!c > 0) {
+  if (!(c > 0)) {
     croak("No extensions found\n");
   } else if (i >= c || i < 0) {
     croak("Requested extension index out of range\n");
-- 
1.9.3

```

**Comment by dsully** [comment]
Thanks, can you submit a GitHub pull request?

**Comment by ppisar** [comment]
On Wed, Feb 11, 2015 at 04:41:11AM -0800, Dan Sully wrote:
> Thanks, can you submit a GitHub pull request?
> 
I can't. The patch has been posted to this this thread.

-- Petr

**Comment by dsully** [comment]
Why can't you? You obviously have a GitHub account.

**Comment by ppisar** [comment]
On Wed, Feb 11, 2015 at 05:28:11AM -0800, Dan Sully wrote:
> Why can't you? You obviously have a GitHub account.
> 
I don't like the github license. Thus I do not share any code there. Also the
web interface full of javascript does not taste me.

-- Petr

**Comment by dsully** [comment]
Not very practical in my opinion. I suggest you reconsider your stance.

**Comment by legart** [comment]
I can see that the fix has been applied, it is possible you could make a new release? Thanks!

**Comment by zebardy** [comment]
Can I also add myself to requesting a new release containing this patch. Many thanks.

**Comment by jonasbn** [comment]
This was fixed with release 1.8.7

PR: #43 from Bernhard M. Wiedemann ( @bmwiedemann )
